src/catalog/column_catalog.cpp

```diff
@@ -21,29 +21,57 @@ namespace peloton {
 namespace catalog {
 
 ColumnCatalogObject::ColumnCatalogObject(executor::LogicalTile *tile,
-                                         int tupleId)
-    : table_oid(tile->GetValue(tupleId, ColumnCatalog::ColumnId::TABLE_OID)
-                    .GetAs<oid_t>()),
-      column_name(tile->GetValue(tupleId, ColumnCatalog::ColumnId::COLUMN_NAME)
-                      .ToString()),
-      column_id(tile->GetValue(tupleId, ColumnCatalog::ColumnId::COLUMN_ID)
-                    .GetAs<oid_t>()),
-      column_offset(
-          tile->GetValue(tupleId, ColumnCatalog::ColumnId::COLUMN_OFFSET)
-              .GetAs<oid_t>()),
-      column_type(StringToTypeId(
-          tile->GetValue(tupleId, ColumnCatalog::ColumnId::COLUMN_TYPE)
-              .ToString())),
-      is_inlined(tile->GetValue(tupleId, ColumnCatalog::ColumnId::IS_INLINED)
-                     .GetAs<bool>()),
-      is_primary(tile->GetValue(tupleId, ColumnCatalog::ColumnId::IS_PRIMARY)
-                     .GetAs<bool>()),
-      is_not_null(tile->GetValue(tupleId, ColumnCatalog::ColumnId::IS_NOT_NULL)
-                      .GetAs<bool>()) {}
-
-ColumnCatalog *ColumnCatalog::GetInstance(storage::Database *pg_catalog,
-                                          type::AbstractPool *pool,
-                                          concurrency::TransactionContext *txn) {
+                                         int tupleId) {
+  auto field_location =
+      tile->GetTuplePointer(tupleId, ColumnCatalog::ColumnId::TABLE_OID);
+  PL_ASSERT(field_location != nullptr);
+  table_oid = *(reinterpret_cast<const oid_t *>(field_location));
+
+  field_location =
+      tile->GetTuplePointer(tupleId, ColumnCatalog::ColumnId::COLUMN_NAME);
+  PL_ASSERT(field_location != nullptr);
+  auto varchar = *(reinterpret_cast<const char *const *>(field_location));
```

Comment:
Quick question. 
Does Peloton store all the varchar type value in place or using a char * pointer as a placeholder or both?